pandas/core/arrays/masked.py

```diff
@@ -159,24 +168,29 @@ def _hasna(self) -> bool:
         # source code using it..
         return self._mask.any()
 
-    def isna(self):
+    def isna(self) -> np.ndarray:
         return self._mask
 
     @property
     def _na_value(self):
         return self.dtype.na_value
 
     @property
-    def nbytes(self):
+    def nbytes(self) -> int:
         return self._data.nbytes + self._mask.nbytes
 
     @classmethod
-    def _concat_same_type(cls, to_concat):
+    def _concat_same_type(cls: Type[BaseMaskedArrayT], to_concat) -> BaseMaskedArrayT:
```

Comment:
we need to add the typevar to avoid...

pandas\core\arrays\integer.py:117: error: Incompatible return value type (got "BaseMaskedArray", expected "IntegerArray")
pandas\core\arrays\boolean.py:122: error: Incompatible return value type (got "BaseMaskedArray", expected "BooleanArray")

we can't use the unbound typevar from pandas._typing here otherwise we get...

pandas\core\arrays\masked.py:183: error: Too many arguments for "object"

since the typevar is needed here, it is also used for the other methods that return type(self)

Reply:
As I said in another issue, I don't have problems with it if there is no way around it, but it needs to be documented then

Reply:
I don't think this is a problem; it's documented in PEP 484

https://www.python.org/dev/peps/pep-0484/#annotating-instance-and-class-methods